sway/input/keyboard.c

```diff
@@ -9,88 +9,92 @@
 #include "sway/commands.h"
 #include "log.h"
 
-static bool keysym_is_modifier(xkb_keysym_t keysym) {
-	switch (keysym) {
-	case XKB_KEY_Shift_L: case XKB_KEY_Shift_R:
-	case XKB_KEY_Control_L: case XKB_KEY_Control_R:
-	case XKB_KEY_Caps_Lock:
-	case XKB_KEY_Shift_Lock:
-	case XKB_KEY_Meta_L: case XKB_KEY_Meta_R:
-	case XKB_KEY_Alt_L: case XKB_KEY_Alt_R:
-	case XKB_KEY_Super_L: case XKB_KEY_Super_R:
-	case XKB_KEY_Hyper_L: case XKB_KEY_Hyper_R:
-		return true;
-	default:
-		return false;
-	}
-}
-
-static size_t pressed_keysyms_length(xkb_keysym_t *pressed_keysyms) {
-	size_t n = 0;
-	for (size_t i = 0; i < SWAY_KEYBOARD_PRESSED_KEYSYMS_CAP; ++i) {
-		if (pressed_keysyms[i] != XKB_KEY_NoSymbol) {
-			++n;
+/**
+ * Update the shortcut model state in response to new input
+ */
+static void update_shortcut_model(struct sway_shortcut_state *state,
```

Comment:
I don't really like the use of "model", which makes it sound more complicated than it is. What about something like `update_shortcut_state`?

Reply:
That might work, if not for the existence of the function `check_shortcut_model` which also acts on the shortcut state.  Then again, the shortcut model is the rule set relating key input, binding output, and the state, so `check_shortcut_model` isn't strictly correct either.

The variations `struct sway_shortcut_model_state`,  `update_shortcut_model_state`, `get_bindings_matching_shortcut_model_state` are more precise, but verbose. 

I'll think about this again in eight hours.

Reply:
`update_shortcut_model` stays, because it's the most apt description I could find. The shortcut model is a reduction of the full key history; and as given by the function comment, `update_shortcut_model` does precisely that. I did rename `check_shortcut_model` to `get_active_binding`.

Reply:
Done.